source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -3,17 +3,53 @@
 #include "source/common/common/assert.h"
 #include "source/common/common/enum_to_int.h"
 #include "source/common/common/logger.h"
+#include "source/common/config/xds_resource.h"
 #include "source/common/http/codes.h"
+#include "source/common/http/utility.h"
+#include "source/extensions/filters/http/well_known_names.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace OnDemand {
 
+namespace {
+
+template <typename ProtoConfig>
+Upstream::OdCdsApiHandlePtr createOdCdsApi(const ProtoConfig& proto_config,
+                                           Upstream::ClusterManager& cm,
+                                           ProtobufMessage::ValidationVisitor& validation_visitor) {
+  if (!proto_config.has_odcds_config() && proto_config.resources_locator().empty()) {
```

Comment:
IIRC the config-source is required no matter if resources-locator is set or not.
@htuch to correct me if I'm wrong.

Reply:
Sorry, my confusion here. I was reading this as resource-name instead of locator.